### PR TITLE
Relax checkstyle rules for javadocs

### LINKED ISSUE #1113: Relax Checkstyle Javadoc rules
The current Javadoc rules enforced by Checkstyle are too strict. There is no need to define docs for a logger field, setter/getters that are too obvious on beans, or empty ctors that dont do anything special. Relax the rules. Proposed by @leleuj

### PR DESCRIPTION
**Pull request by mmoayyed**
Closes #1113

### CODE REVIEW

File: checkstyle-rules.xml [hunk 1]
```diff
@@ -50,6 +52,7 @@ Checkstyle configuration based on Sun's conventions, compliant with CAS coding c
     <module name="JavadocVariable">
       <property name="severity" value="error"/>
       <property name="scope" value="protected" />
+      <property name="ignoreNamePattern" value="log|logger|LOG|LOGGER"/>
```
Comment (leleuj) [inline checkstyle-rules.xml]: It means we don't need to comment for loggers, doesn't we?

Comment (leleuj) [inline checkstyle-rules.xml]: But we still need to comment any property? Like `status` or `formData`...

Comment (mmoayyed) [inline checkstyle-rules.xml]: No comment required for loggers, yes. Properties can be ignored as long as they are conformant to Java beans. So for example, getStatus and setStatus do need docs, but getAuthenticationTokenBasedOnX() would need one. The property javadoc setting is controlled via `<property name="allowMissingPropertyJavadoc" value="true"/>`

Comment (leleuj) [inline checkstyle-rules.xml]: OK

**Pull request merged**